### PR TITLE
refactor: logger

### LINKED ISSUE #294: refactor logger
Convert it from static to a dependency, and import it in the modules

### PR DESCRIPTION
**Pull request by tgntr**
**Detailed description**:
Introduce logger as a dependency and only do critical logging(panic) in api modules
**Which issue(s) this PR fixes**:
Fixes #294 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: execution/extrinsic/runtime_extrinsic_test.go [hunk 1]
```diff
@@ -214,8 +215,10 @@ func Test_RuntimeExtrinsic_CheckInherents(t *testing.T) {
 	mockModuleOne.On("CheckInherent", mockCallOne, inherentData).Return(nil)
 	mockModuleTwo.On("IsInherent", mockCallOne).Return(false)
 
-	result := target.CheckInherents(inherentData, mockBlock)
+	result, err := target.CheckInherents(inherentData, mockBlock)
+	assert.NoError(t, err)
 
+	assert.NoError(t, err)
```
Comment (failfmi) [inline execution/extrinsic/runtime_extrinsic_test.go]: nit: no need

File: execution/extrinsic/runtime_extrinsic_test.go [hunk 2]
```diff
@@ -244,8 +247,10 @@ func Test_RuntimeExtrinsic_CheckInherents_FatalError(t *testing.T) {
 	mockModuleOne.On("CheckInherent", mockCallOne, inherentData).Return(err)
 	mockModuleOne.On("InherentIdentifier").Return(inherentIdentifier)
 
-	result := target.CheckInherents(inherentData, mockBlock)
+	result, errCheckInherents := target.CheckInherents(inherentData, mockBlock)
+	assert.NoError(t, errCheckInherents)
 
+	assert.NoError(t, errCheckInherents)
```
Comment (failfmi) [inline execution/extrinsic/runtime_extrinsic_test.go]: nit: no need

File: execution/extrinsic/runtime_extrinsic_test.go [hunk 3]
```diff
@@ -267,8 +272,10 @@ func Test_RuntimeExtrinsic_CheckInherents_NoInherents(t *testing.T) {
 	mockModuleOne.On("IsInherent", mockCallOne).Return(false)
 	mockModuleTwo.On("IsInherent", mockCallOne).Return(false)
 
-	result := target.CheckInherents(inherentData, mockBlock)
+	result, err := target.CheckInherents(inherentData, mockBlock)
+	assert.NoError(t, err)
 
+	assert.NoError(t, err)
```
Comment (failfmi) [inline execution/extrinsic/runtime_extrinsic_test.go]: nit: no need

File: execution/extrinsic/runtime_extrinsic_test.go [hunk 4]
```diff
@@ -286,8 +293,10 @@ func Test_RuntimeExtrinsic_CheckInherents_Signed(t *testing.T) {
 	mockBlock.On("Extrinsics").Return(sc.Sequence[primitives.UncheckedExtrinsic]{mockUncheckedExtrinsic})
 	mockUncheckedExtrinsic.On("IsSigned").Return(true)
 
-	result := target.CheckInherents(inherentData, mockBlock)
+	result, err := target.CheckInherents(inherentData, mockBlock)
+	assert.NoError(t, err)
 
+	assert.NoError(t, err)
```
Comment (failfmi) [inline execution/extrinsic/runtime_extrinsic_test.go]: nit: no need

File: primitives/log/logger.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+type DebugLogger interface {
+	TraceLogger
+	Debug(message string)
+	Debugf(message string, a ...any)
+}
+
+type WarnLogger interface {
+	DebugLogger
+	Info(message string)
+	Infof(message string, a ...any)
```
Comment (failfmi) [inline primitives/log/logger.go]: Why are these here?

File: execution/extrinsic/runtime_extrinsic_test.go [hunk 6]
```diff
@@ -244,8 +247,10 @@ func Test_RuntimeExtrinsic_CheckInherents_FatalError(t *testing.T) {
 	mockModuleOne.On("CheckInherent", mockCallOne, inherentData).Return(err)
 	mockModuleOne.On("InherentIdentifier").Return(inherentIdentifier)
 
-	result := target.CheckInherents(inherentData, mockBlock)
+	result, errCheckInherents := target.CheckInherents(inherentData, mockBlock)
```
Comment (failfmi) [inline execution/extrinsic/runtime_extrinsic_test.go]: `errCheckInherents` -> `err` and rename the current `err` to something else

File: frame/executive/executive_test.go [hunk 7]
```diff
@@ -420,6 +424,49 @@ func Test_Executive_ApplyExtrinsic_Success_DispatchOutcomeNil(t *testing.T) {
...
+	mockUncheckedExtrinsic.On("Check").Return(mockCheckedExtrinsic, nil)
+	mockSystemModule.On("NoteExtrinsic", mockUncheckedExtrinsic.Bytes())
+
+	mockCheckedExtrinsic.On("Function").Return(mockCall)
+	mockCall.On("BaseWeight").Return(baseWeight)
+	mockCall.On("WeighData", baseWeight).Return(dispatchInfo.Weight)
+	mockCall.On("ClassifyDispatch", baseWeight).Return(primitives.DispatchClass{VaryingData: sc.NewVaryingData()})
+	mockCall.On("PaysFee", baseWeight).Return(dispatchInfo.PaysFee)
+	mockCheckedExtrinsic.On("Apply", unsignedValidator, mock.Anything, encodedExtrinsicLen).
+		Return(*dispatchResultOk, nil)
```
Comment (failfmi) [inline frame/executive/executive_test.go:442]: No mock assertions after execution.

**Review by failfmi — DISMISSED** [review]

**Review by radkomih — DISMISSED** [review]

**Review by failfmi — APPROVED** [review]

**Review by radkomih — APPROVED** [review]
